models/Settings.php

```diff
@@ -28,14 +28,14 @@ class Settings extends Model
 
     public function initSettingsData()
     {
-        $this->require_activation = true;
-        $this->activate_mode = self::ACTIVATE_AUTO;
-        $this->use_throttle = true;
-        $this->block_persistence = false;
-        $this->allow_registration = true;
-        $this->login_attribute = self::LOGIN_EMAIL;
-        $this->remember_login = self::REMEMBER_ALWAYS;
-        $this->use_register_throttle = true;
+        $this->require_activation = config('rainlab.user::requireActivation');
+        $this->activate_mode = config('rainlab.user::activateMode');
+        $this->use_throttle = config('rainlab.user::useThrottle');
+        $this->block_persistence = config('rainlab.user::blockPersistence');
+        $this->allow_registration = config('rainlab.user::allowRegistration');;
+        $this->login_attribute = config('rainlab.user::loginAttribute');
+        $this->remember_login = config('rainlab.user::rememberLogin');
+        $this->use_register_throttle = config('rainlab.user::useRegisterThrottle');
```

Comment:
@bennothommo what do you think about not including the defaults here? For October we have to always make sure that config calls include their defaults because updates don't include config files. However, plugin updates would include the updated config files, so we could technically get by without including the defaults here.

I'm personally leaning more towards including the defaults in the config file and here, simply to encourage that to be the standard practice, even if it does mean that if we want to change the defaults we would have to change them in two separate places.

Reply:
I like where you're coming from @LukeTowers, this plugin has sort of become a best-practices example. That said, I don't think adding the extra default value for that purpose is a good idea here. I feel like it would be misleading, and create an ambiguous source of truth. The default values provided in the model would be unreachable code.

While having default values there is a best practice _when there is no file-based configuration_, I would argue that omitting them is a better practice when there is.

Reply:
Fair enough, it's true enough that as far as plugin development goes, so long as the plugin includes the config file then the default value shouldn't be included in the code, which should be all the time. With core development it's different, since you can guarantee that there will be users without the updated config files.

I do still think there's an argument to made for setting the example of including defaults in code though, since it's easily possible for a developer to forget to include a default config file, and including the defaults in code means that you're always safe from that situation.

Reply:
If we're that concerned about people using this plugin as a verbatim learning resource, would a comment explaining where the default values live be sufficient? Either way, just let me know what you need from me to make this mergeable 🙂

Reply:
@LukeTowers @scottbedard Even if it's unreachable, it's a visual indicator of what the default is and it kinda is a standard used elsewhere, so I feel it best to include it.

Reply:
@LukeTowers @bennothommo Additional values have been added, let me know if there is anything else you need from me. Thanks guys!